snapcraft/commands/extensions.py

```diff
@@ -135,5 +134,5 @@ def run(self, parsed_args):
         # not part of the Project model
         extract_parse_info(yaml_data_for_arch)
 
-        models.Project.unmarshal(yaml_data_for_arch)
-        emit.message(yaml.safe_dump(yaml_data_for_arch, indent=4, sort_keys=False))
+        project_data = models.Project.unmarshal(yaml_data_for_arch)
+        emit.message(project_data.to_yaml_string())
```

Comment:
This changes the order of the yaml file to match how fields are defined in the model.

Previously, the order was not changed because the data was not unmarshalled and marshalled. Now that the validators for `platforms` and `architectures` are populating the dictionary with the right data types, it's now possible to unmarshal and marshal.